### ISSUE TITLE
Possible LLVM miscompilation with `ReleaseFast`

### ISSUE THREAD

**Issue by henrikolsen549** [ziglang/zig#17023]
### Zig Version

0.11.0, 0.12.0-dev.185+49075d205

### Steps to Reproduce and Observed Behavior

See this [Zig source file](https://gist.github.com/henrikolsen549/04280703d68b3b6603c9a2bcf9cd30b6) (`main.zig`).

Correct output when run with Debug (`zig run main.zig -O Debug`):
```
item: inner
obj len 1
item: inner
```
Incorrect output when run with ReleaseFast (`zig run main.zig -O ReleaseFast`):
```
item: inner
obj len 1
item:
```

The bug disappears with a sufficiently low value of `-fopt-bisect-limit`:
`zig run main.zig -O ReleaseFast -fopt-bisect-limit=90000`

### Expected Behavior

`ReleaseFast` should be functionally equivalent to `Debug`.

**Comment by Techatrix** [comment]
could be related to #16392. I also reduced that issue from using `std.json.parseFromValue`.

**Comment by henrikolsen549** [comment]
Fixed by #17963.
